### DISCUSSION TITLE
Attempting to attack an invalid entity

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#1179] [Help] [answered]
So I have the following code
```java
EntityProjectile fireball = new EntityProjectile(event.getPlayer(), EntityType.FIREBALL);
Pos position = event.getPlayer().getPosition();
Vec direction = event.getPlayer().getPosition().direction();
fireball.setVelocity(direction.normalize().mul(20));
fireball.setGravity(0.0, 0.0);
fireball.setInstance(event.getPlayer().getInstance(), position.add(0, event.getPlayer().getEyeHeight(),0));
```

But if a player manages to hit a fireball back it kicks them with `Attempting to attack an invalid entity`, what should I do about this?

Console error: `(AttackManager.entityHit) - ERROR - Player xyz tried to attack invalid mob`

**Comment by para-dise** [comment, marked as answer]
@TogAr2 I noticed this issue come from MinestomPvP and not minestom itself, any ideas?

**Reply by TogAr2** [reply]
It should be fixed in the latest MinestomPvP commit

**Reply by para-dise** [reply]
Yep that fixed it, tyvm
